### ISSUE TITLE
🐛 CSS parser broken on main

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#9253]
### Environment information

main

### What happened?

```css
@container   scroll-state(  scrolled:   bottom  )   {  }
```

The code currently break with a SCSS diagnostics, even though it was supposed be parsed as CSS. 

```
  message: Expected a SCSS expression but instead found ':   bottom'.,
  advice: ParserAdvice {
      advice_list: [
          Detail(
              ParserAdviceDetail {
                  message: "Expected a SCSS expression here.",
                  span: Some(
                      37..47,
                  ),
              },
          ),
      ],
  },
```

And then the formatter breaks printing 

```
quick_test:1:37 parse ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ✖ Unexpected value or character.
  
  > 1 │ @container   scroll-state(  scrolled:   bottom  )   {  }
      │                                     ^^^^^^^^^^
    2 │ 
  
  ℹ Expected one of:
```

@denbezrukov I assign this to you because you're working on it. I'm worried if we do a release, it will break people's code

### Expected result

It shouldn't break and it shouldn't show SCSS parsing errors

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
